src/main/java/ListNode.java

```diff
@@ -0,0 +1,44 @@
+public class ListNode {
+    public ListNode next;
+    public int data;
+
+    public ListNode(int data) {
+        this.data = data;
+    }
+
+    public static ListNode reverseLinkedList(ListNode head) {
+        ListNode prev = null;
+        ListNode next = null;
+        ListNode cur = head;
+        while(cur != null) {
+            next = cur.next;
+            cur.next = prev;
+            prev = cur;
+            cur = next;
+        }
+        return prev;
+    }
+
+    public static ListNode mergeLinkedLists(ListNode listA, ListNode listB) {h
+        ListNode head = new ListNode();
+        mergeLinkedLists(head, listA, listB);
+        return head.next;
+    }
+
+    private static void mergeLinkedLists(ListNode merged, ListNode a, ListNode b) {
+        if(a == null) {
+            merged.next = b;
+        } else if(b == null) {
+            merged.next = a;
+        } else {
+            if(a.data < b.data) {
+                merged.next = a
+                mergeLinkedLists(merged, a.next, b);
```

Comment:
This isn't really a necessary case for recursion; you typically use recursion when there's a bunch of state that you just want to push onto the stack and pop off once you're done. In this case, you're literally iterating over pulling each item off the list, so recursion is a bit of overkill here.